tests/github_actions/checks/test_extra_checks.py

```diff
@@ -0,0 +1,79 @@
+from pathlib import Path
+
+from checkov.github_actions.runner import Runner
+from checkov.runner_filter import RunnerFilter
+
+EXTRA_YAML_CHECKS_DIR = Path(__file__).parent / "extra_yaml_checks"
+RESOURCES_DIR = Path(__file__).parent.parent / "resources"
+
+
+def test_simple_attribute_check():
+    # given
+    test_file = str(RESOURCES_DIR / ".github/workflows/slsa-gen.yaml")
+    runner = Runner()
+
+    # when
+    report = runner.run(
+        files=[test_file],
+        external_checks_dir=[str(EXTRA_YAML_CHECKS_DIR)],
+        runner_filter=RunnerFilter(checks="CKV2_GHA_CUSTOM_1"),
+    )
+
+    # remove all checks
+    runner.graph_registry.checks.clear()
+
+    # then
+    summary = report.get_summary()
+
+    passing_resources = {
+        "jobs.build",
+        "jobs.scan",
+    }
+    failing_resources = {
+        "jobs.attest",
+        "jobs.provenance",
+    }
+
+    passed_check_resources = {c.resource for c in report.passed_checks}
+    failed_check_resources = {c.resource for c in report.failed_checks}
+
+    assert summary["passed"] == len(passing_resources)
+    assert summary["failed"] == len(failing_resources)
+    assert summary["skipped"] == 0
+    assert summary["parsing_errors"] == 0
+
+    assert passed_check_resources == passing_resources
+    assert failed_check_resources == failing_resources
+
+
+def test_jobs_steps_connection_check():
+    # given
+    test_file = str(RESOURCES_DIR / ".github/workflows/slsa-gen.yaml")
+    runner = Runner()
+
+    # when
+    report = runner.run(
+        files=[test_file],
+        external_checks_dir=[str(EXTRA_YAML_CHECKS_DIR)],
+        runner_filter=RunnerFilter(checks="CKV2_GHA_CUSTOM_2"),
+    )
+
+    # remove all checks
+    runner.graph_registry.checks.clear()
+
+    # then
+    summary = report.get_summary()
+
+
+    failing_resources = {
+        "jobs.attest.steps.5",
+    }
```

Comment:
Is that compatible with the resource ids already generated for steps with the python checks? I think that the format for step resources should be 'jobs.<job_name>.steps.<step_name>'

Reply:
the `name` field is optional and you can name all your steps the same, so if we use this in Python checks, we probably should reconsider it. only the order is relevant at the end.